packages/lambda_streams/lambda_streams.0.1.1/opam

```diff
@@ -0,0 +1,42 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+version: "0.1.2"
```

Comment:
This os redundant, opam will override it with the value in the path, and in this case emit a warning as they don’t match

Reply:
Can you clarify a bit more? do you mean the `opam-version` or the `version` or both? opam will override which value in the path? what doesn't match? Sorry for all the questions, I read the opam and dune docs but I didn't find anything written about the stuff mentioned here.

The `opam-version` is generated by `dune-project`. I added the project version because it was a main stanza for `dune-project` and I was just assuming it's added to the auto-generated `.opam` file for the same reasons as other package managers like npm's `package.json` or rust's `Cargo.toml`

Reply:
The version field. Opam infers the version of the package from the path in the opam repository, in this case it gets 0.1.1, and uses that as the package version. Here you set 0.1.2 which should give rise to a warning. To change the version you need to git mv the opam file to `packages/lambda_streams/lambda_streams.0.1.2/opam`

```suggestion
```

Reply:
I have double checked. If you open the build logs, currently you read
```
[ERROR] In the opam file for lambda_streams.0.1.1:
          - At /home/opam/.opam/repo/default/packages/lambda_streams/lambda_streams.0.1.1/opam:3:0:
            This file is for version '0.1.1' but its 'version:' field advertises '0.1.2'.
        'version' has been ignored.
```

Reply:
Ah I forgot to update the folder name, thanks for catching that. So is it best practices to leave out the version field in `dune-project`/`opam`? or just that you don't necessarily need it for publishing to opam?

Reply:
Unless you need it for some specific reason is best left out, otherwise we just delete it when pushing to the opam repo